# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: third_party/rust/naga/src/front/spv/mod.rs
# Commit: b3109ad14685
# Full Hash: b3109ad14685624ada7861083ff5332aec9a7818
# Author: Andy Leiserson <aleiserson@mozilla.com>
# Date: 2025-10-07 17:00 +0000
# Description:
#   Bug 1992842 - Update wgpu to upstream 1072b878 (2025-10-06) r=webgpu-reviewers,supply-chain-reviewers,ErichDonGubler
# ==============================================================================

diff -r 1d3f000cf1f9 -r b3109ad14685 third_party/rust/naga/src/front/spv/mod.rs
--- a/third_party/rust/naga/src/front/spv/mod.rs	Tue Oct 07 16:51:44 2025 +0000
+++ b/third_party/rust/naga/src/front/spv/mod.rs	Tue Oct 07 17:00:40 2025 +0000
@@ -44,7 +44,6 @@
 use super::atomic_upgrade::Upgrades;
 use crate::{
     arena::{Arena, Handle, UniqueArena},
-    path_like::PathLikeOwned,
     proc::{Alignment, Layouter},
     FastHashMap, FastHashSet, FastIndexMap,
 };
@@ -384,7 +383,7 @@
     pub adjust_coordinate_space: bool,
     /// Only allow shaders with the known set of capabilities.
     pub strict_capabilities: bool,
-    pub block_ctx_dump_prefix: Option<PathLikeOwned>,
+    pub block_ctx_dump_prefix: Option<String>,
 }
 
 impl Default for Options {
@@ -5636,7 +5635,7 @@
         let is_depth = self.next()?;
         let is_array = self.next()? != 0;
         let is_msaa = self.next()? != 0;
-        let _is_sampled = self.next()?;
+        let is_sampled = self.next()?;
         let format = self.next()?;
 
         let dim = map_image_dim(dim)?;
@@ -5671,6 +5670,8 @@
                     format: map_image_format(format)?,
                     access: crate::StorageAccess::default(),
                 }
+            } else if is_sampled == 2 {
+                return Err(Error::InvalidImageWriteType);
             } else {
                 crate::ImageClass::Sampled {
                     kind,